### PR TITLE
device: Added feature to support Sharp printers

### PR DESCRIPTION
**Pull request by mhenreaux**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

### CODE REVIEW

File: includes/definitions.inc.php [hunk 1]
```diff
@@ -1538,6 +1538,14 @@
 $config['os'][$os]['over'][0]['graph'] = 'device_toner';
 $config['os'][$os]['over'][0]['text']  = 'Toner';
 
+$os = 'sharpprinter';
```
Comment (laf) [inline includes/definitions.inc.php]: This should be just `sharp`

File: includes/discovery/os/sharpprinter.inc.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?php
+if (!$os) {
+    // Sharp Multifunction Printer/Scanner
+    if (strstr($sysDescr, 'SHARP MX-')) {
```
Comment (laf) [inline includes/discovery/os/sharpprinter.inc.php]: We have custom functions to deal with this, you should update this to:

`if (str_contains($sysDescr, 'SHARP MX-')) {`

**Comment by laf** [comment]
Thanks, I'd say we need to rename this OS to `sharp` to be more consistent.

We also need an OS test adding, I've created a patch which you can use:

https://gist.github.com/laf/bda3cda8e9decc0372c055cb06af62fd

Ideally you should put the full sysDescr in that OS check if you have it.

After that, all good to go and thanks for contributing :)

**Comment by mhenreaux** [comment]
Hello @laf ,

I've extracted the sysDescr of tested printers:
iso.3.6.1.2.1.1.1.0 = STRING: "SHARP MX-2614N"
iso.3.6.1.2.1.1.1.0 = STRING: "SHARP MX-C301W"
iso.3.6.1.2.1.1.1.0 = STRING: "SHARP MX-3140N"

Regards,
Morgan

**Comment by laf** [comment]
I'm merging this but will submit a change straight after to fix the issues mentioned.

It's highly recommended you don't use your master branch for submitting changes - see our docs for more info.

**Comment by laf** [comment]
See https://github.com/librenms/librenms/pull/4452 for the changes made.

**Pull request merged**